### DISCUSSION TITLE
Frictional contact

### DISCUSSION THREAD

**Discussion by samyogshr** [idaholab/moose#17264] [Q&A Modules: General] [answered]
Hi Moose users,

I am modeling a 3D structural dynamics problem (using Hex8 elements) with frictional contact (penalty formulation) in MOOSE. The analysis converges and displacements look fine. But, there are spikes (high frequency oscillation at few time-steps) in acceleration output at some nodes in the slave and master surfaces (specially prominent at corners). I came across a few parameters in the contact block that were previously suggested in previous discussions, e.g., tangential_tolerance, normal_smooting_distance etc.
Has anybody else experienced similar problem? Does anybody know if using these parameters will help?
I also couldn't find information on what it means by defining tangential_tolerance = 1e-5 compared to 1e-3 for example. Could somebody please explain or provide a good reference?

Thank you,
Samyog

**Comment by dzheng26** [comment]
@samyogshr , I cannot really help you since I'm also trying to use the 3D frictional contact. I cannot even get it converge for my example. Could you post your .i file here so that it may help me get my example converged first. Many Thanks.

**Reply by samyogshr** [reply]
@dzheng-conti , hope you have resolved it by now. Sorry for late reply.

Contact block in my input file. I have 5 surfaces in contact and using 8-noded brick elements.

[Contact]
  [./base]
    secondary = 10006
    primary = 105
    model = coulomb
    formulation = penalty
    normalize_penalty = true
    friction_coefficient = 0.2
tangential_tolerance = 1e-5    
penalty = 1e4
normal_smoothing_distance = 0.1    
  [../]

  [./side2]
    secondary = 10008
    primary = 107
   model = coulomb
    formulation = penalty
    normalize_penalty = true
    friction_coefficient = 0.2
tangential_tolerance = 1e-5  
  penalty = 1e4
normal_smoothing_distance = 0.1    
  [../]

[./side3]
    secondary = 10010
    primary = 109
   model = coulomb
    formulation = penalty
    normalize_penalty = true
    friction_coefficient = 0.2
tangential_tolerance = 1e-5  
  penalty = 1e4
normal_smoothing_distance = 0.1    
  [../]

[./side4]
    secondary = 10012
    primary = 111
   model = coulomb
    formulation = penalty
    normalize_penalty = true
    friction_coefficient = 0.2
tangential_tolerance = 1e-5  
  penalty = 1e4
normal_smoothing_distance = 0.1    
  [../]

[./side5]
    secondary = 10014
    primary = 113
   model = coulomb
    formulation = penalty
    normalize_penalty = true
    friction_coefficient = 0.2
tangential_tolerance = 1e-5  
  penalty = 1e4
normal_smoothing_distance = 0.1    
  [../]
[]

**Reply by dzheng26** [reply]
Thanks for your help. I got my problem somewhat resolved. It converged but slowly. The contact works better with lower friction coefficient. In my case, the friction coefficient is 1.0

**Comment by GiudGiud** [comment, marked as answer]
@bwspenc should be able to help

**Reply by samyogshr** [reply]
@bwspenc Could you please share your thoughts on this, thank you.

**Reply by bwspenc** [reply]
Sorry for coming into this way too late!  The `tangential_tolerance` parameter controls the distance (in the physical coordinates of your model) that the face is virtually extended beyond the edge. This helps in problems where you have two blocks that are in contact right along their edges. I'm not sure whether that would help your particular problem.

To prevent high-frequency noise, anything you can do to smooth the behavior of contact will help. `normal_smoothing_distance` should help somewhat with that, as would a lower friction coefficient, as you already found.

**Reply by hugary1995** [reply]
In addition to Ben's suggestions, you could also incorporate Rayleigh damping. You can use a higher damping coefficient near the contact surfaces.
